src/Symfony/Component/Translation/Bridge/Phrase/README.md

```diff
@@ -0,0 +1,106 @@
+# Phrase Translation Provider
+
+Symfony phrase translation provider bridge
+
+## DSN example
+
+```dotenv
+PHRASE_DSN=phrase://PROJECT_ID:API_TOKEN@default?userAgent=myProject
+```
+
+### DSN elements
+
+- `PROJECT_ID`: can be retrieved in phrase from `project settings > API > Project ID`
+- `API_TOKEN`: can be created in your [phrase profile settings](https://app.phrase.com/settings/oauth_access_tokens)
+- `default`: endpoint, defaults to `api.phrase.com`
+
+### Required dsn query parameters
+
+- `userAgent`: please read [this](https://developers.phrase.com/api/#overview--identification-via-user-agent) for some examples.
+
+See [fine tuning your phrase api calls](#fine-tuning-your-phrase-api-calls) for additional DSN options.
+
+## Phrase locale names
+
+Translations being imported using the symfony xlf format in phrase, locales are matched on locale name in phrase.
+Therefor it's necessary the locale names should be as defined in [RFC4646](https://www.ietf.org/rfc/rfc4646.txt) (e.g. pt-BR rather than pt_BR).
+Not doing so will result in phrase creating a new locale for the imported keys.
+
+## Locale creation
+
+If you define a locale in your `translation.yaml` which is not configured in your phrase project, it will be automatically created. Deletion of locales however, is (currently) not managed by this provider.
+
+## Domains as tags
+
+Translations will be tagged in phrase with the symfony translation domain they belong to.
+Check the [wickedone/phrase-translation-bundle](https://github.com/wickedOne/phrase-translation-bundle) if you need help managing your tags in phrase.
+
+## Cache
+
+The read responses from phrase are cached to speed up the read and delete method of this provider and also to contribute to the rate limit as little as possible.
+Therefor the factory should be initialised with a PSR-6 compatible cache adapter.
+
+## Events
+
+To enable you to perform post-processing on translation values and / or keys, two events are dispatched by this provider class.
```

Comment:
Can you explain why you added those events?
They introduce a dependency on some event-dispatcher, which should be optional I suppose
Also, shouldn't this capability be made available to all bridges?
I'd be fine splitting this concern to another more generic PR.

Reply:
Let's move this to another PR then.

Reply:
to be sure: you want the event dispatcher to be optional and i assume configurable?

Reply:
yes, but in another PR. In this PR, I would remove everything related to it.

Reply:
https://github.com/symfony/symfony/issues/49520